include/envoy/config/discovery_service_base.h

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include "envoy/config/subscription.h"
+
+namespace Envoy {
+namespace Config {
+
+class RdsRouteConfigSubscriptionBase : public Config::SubscriptionCallbacks {};
```

Comment:
I don't think we want  to move all of these APIs into a single header here. Ideally these are localized and grouped with their logical part of the code base (e.g. listener, router, etc.).

Reply:
@htuch It is good for me to add `getResourceName()` on any base class of Config Subscription.

Reply:
I think each subscription just needs to provide its latest API (v3) resource proto descriptor or type name. From this, we can programatically infer (1) earlier resource versions via the UDPA previous type breadcrumb annotation and (2) the service endpoints via the `service` resource type annotations (since we have a backmap).